### PR TITLE
chore: fix publishing with beachball

### PR DESCRIPTION
**Pull request by layershifter**
This PR fixes publishing with `beachball`, see comments on files themselves for details.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -27,9 +27,6 @@ steps:
   - script: yarn install --frozen-lockfile
     displayName: Install dependencies
 
-  - script: yarn nx run-many --target=build --all --parallel --max-parallel=3
-    displayName: Build artifacts
-
```
Comment (layershifter) [inline azure-pipelines.yml:32]: Build nows run in a Beachball hook.

File: beachball.hooks.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  // `beachball` runs this hook for every package, we want to run it once.
+  // If we will run build before `beachball publish` artifacts will have
+  // old (without bump) versions.
+  async prepublish() {
+    if (!completedPostBump) {
+      await sh('yarn build --skip-nx-cache');
+      completedPostBump = true;
+    }
+  },
+};
```
Comment (layershifter) [inline beachball.hooks.js:56]: # Before

```mermaid
graph TD
    A[Build packages] --> B[Bump versions]
    B --> |Try to sync version in the patch| C[Publish to NPM]
```

### After

```mermaid
graph TD
    A[Bump versions] --> B[Build packages]
    B --> C[Publish to NPM]
```

File: packages/jest-serializer/project.json [hunk 3]
```diff
@@ -27,7 +27,14 @@
         "tsConfig": "packages/jest-serializer/tsconfig.lib.json",
         "packageJson": "packages/jest-serializer/package.json",
         "main": "packages/jest-serializer/src/index.ts",
-        "assets": ["packages/jest-serializer/*.md"]
```
Comment (layershifter) [inline packages/jest-serializer/project.json:30]: We don't want to include `CHANGELOG.md` to distributed files.

File: beachball.hooks.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+module.exports = {
+  // `beachball` runs this hook for every package, we want to run it once.
+  // If we will run build before `beachball publish` artifacts will have
+  // old (without bump) versions.
+  async prepublish() {
+    if (!completedPostBump) {
+      await sh('yarn build --skip-nx-cache');
+      completedPostBump = true;
+    }
+  },
```
Comment (miroslavstastny) [inline beachball.hooks.js:55]:
```suggestion
  // Executed after all package versions were bumped -> run build
  // If we run build before `beachball publish`, artifacts would have
  // old (without bump) versions.
  async prepublish() {
    // `beachball` runs this hook for every package, we want to run it only once.
    if (!completedPostBump) {
      await sh('yarn build --skip-nx-cache');
      completedPostBump = true;
    }
  },
```

**Pull request merged**